data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java

```diff
@@ -28,28 +27,23 @@
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
-import org.apache.tuweni.bytes.Bytes32;
-import tech.pegasys.artemis.beaconrestapi.schema.BeaconChainHeadResponse;
-import tech.pegasys.artemis.datastructures.state.BeaconState;
-import tech.pegasys.artemis.datastructures.state.Checkpoint;
+import tech.pegasys.artemis.api.ChainDataProvider;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconChainHead;
 import tech.pegasys.artemis.provider.JsonProvider;
-import tech.pegasys.artemis.storage.ChainStorageClient;
 
 public class BeaconChainHeadHandler implements Handler {
 
-  private final ChainStorageClient client;
+  private final ChainDataProvider chainDataProvider;
```

Comment:
i ended up calling this 'provider' in most places, works across the different classes, and generally the classes only required 1 provider... i'm not overly concerned if we leave this as chainDataProvider, just a comment.